pkg/image/build.go

```diff
@@ -156,6 +167,38 @@ func Build(cfg *config.Config, dir, imageName string, secrets []string, noCache,
 		global.LabelNamespace + "has_init": "true",
 	}
 
+	if isUsingCogBaseImage {
+		labels[global.LabelNamespace + "using_cog_base_image"] = "true"
+		labels[global.LabelNamespace + "cog-base-image-name"] = cogBaseImageName
+
+		// get the last layer of the cog base image so that when we look at the built cog image,
+		// we know where the base image ends
+		// pull the base image, as we'll need to pull it anyway to build the cog image
+		// Q: are we ok with the cog base image being pulled to the user's machine? can fix this later by implementing 
+		// a manifest inspect which doesn't require a pull
+
+		err := docker.Pull(cogBaseImageName)
+		if err != nil {
+			return fmt.Errorf("Failed to pull cog base image: %w", err)
+		}
+
+		cogBaseImage, err := docker.ImageInspect(cogBaseImageName)
+		if err != nil {
+			return fmt.Errorf("Failed to inspect cog base image while trying to fetch last layer: %w", err)
+		}
+
+		if cogBaseImage.RootFS.Layers == nil || len(cogBaseImage.RootFS.Layers) == 0 {
+			return fmt.Errorf("Cog base image has no layers or RootFS is nil: %s", cogBaseImageName)
+		}
+
+		lastLayerIndex := len(cogBaseImage.RootFS.Layers) - 1
+		lastLayer := cogBaseImage.RootFS.Layers[lastLayerIndex]
+		console.Debugf("Last layer of the cog base image: %s", lastLayer) // prints the sha
+
+		labels[global.LabelNamespace + "cog-base-image-last-layer-diff-id"] = lastLayer
```

Comment:
Is this a better name? Or is there some significance to it being a diff.

```suggestion
		labels[global.LabelNamespace + "cog-base-image-last-layer-sha"] = lastLayer
```

Reply:
Yes, it's a diff-id of a layer, rather than the digest of the layer - but it's still a sha. Switched to sha for more conciseness.